docs/proposals/locking.md

```diff
@@ -108,3 +108,198 @@ support:
 |9  |Cater for read-only binary files when merging locally<ul><li>Because files are read-only this might prevent merge from working when actually it's valid.</li><li>Always fine to merge the latest version of a binary file to anywhere else</li><li>Fine to merge the non-latest version if user is aware that this may cause merge problems (see Breaking the rules)</li><li>Therefore this feature is about dealing with the read-only flag and issuing a warning if not the latest</li></ul>|
 |10 |List current locks<ul><li>That the current user has</li><li>That anyone has</li><li>Potentially scoped to folder</li></ul>|`git lfs lock --list [paths...]`
 |11 |Reject a push containing a binary file currently locked by someone else|pre-receive hook on server, allow --force to override (i.e. existing parameter to git push)
+
+## Implementation details
+### Types
+To make the implementing locking on the lfs-test-server as well as other servers
+in the future easier, it makes sense to create a `lock` package that can be
+depended upon from any server. This will go along with Steve's refactor which
+touches the `lfs` package quite a bit.
+
+Below are enumerated some of the types that will presumably land in this
+sub-package.
+
+```go
+// Lock represents a single lock that against a particular path.
+//
+// Locks returned from the API may or may not be currently active, according to
+// the Expired flag.
+type Lock struct {
+        // Path is an absolute path to the file that is locked as a part of this
+        // lock.
+        Path string `json:"path"`
+        // Creator is the author who initiated this lock.
+        Creator struct {
```

Comment:
This is called `Committer` in `LockRequest`, suggest they should be the same for consistency.